src/SFML/Window/WindowBase.cpp

```diff
@@ -146,22 +146,29 @@ bool WindowBase::isOpen() const
 
 
 ////////////////////////////////////////////////////////////
-Event WindowBase::pollEvent()
+std::optional<Event> WindowBase::pollEvent()
 {
-    Event event;
-    if (m_impl && (event = m_impl->popEvent(false)))
-        filterEvent(event);
+    if (m_impl == nullptr)
+        return std::nullopt;
+
+    std::optional<sf::Event> event = m_impl->popEvent(false /* non-blocking */);
+    if (event.has_value())
+        filterEvent(*event);
+
     return event;
 }
 
 
 ////////////////////////////////////////////////////////////
 Event WindowBase::waitEvent()
 {
-    Event event;
-    if (m_impl && (event = m_impl->popEvent(true)))
-        filterEvent(event);
-    return event;
+    assert(m_impl);
+
+    std::optional<sf::Event> event = m_impl->popEvent(true /* blocking */);
+    assert(event.has_value());
```

Comment:
Why `assert` here instead of keeping the original conditional? Can we safely assume `popEvent` never returns `std::nullopt`?

Reply:
Yes, `popEvent` in blocking mode will wait until an event is available. Will add a comment

Reply:
Is it overkill to refactor this into two functions where one returns `Event` and the other `std::optional<Event>`?